src/plat/pc99/machine/ioapic.c

```diff
@@ -118,8 +134,8 @@ exception_t ioapic_decode_map_pin_to_vector(word_t ioapic, word_t pin, word_t le
         current_syscall_error.rangeErrorMax = num_ioapics - 1;
         return EXCEPTION_SYSCALL_ERROR;
     }
-    if (pin >= IOAPIC_IRQ_LINES) {
-        userError("Invalid IOAPIC pin %ld, there are %d pins", (long)pin, IOAPIC_IRQ_LINES);
+    if (pin >= ioapic_nirqs[ioapic]) {
+        userError("Invalid IOAPIC pin %ld, there are %d pins", (long)pin, ioapic_nirqs[ioapic]);
         current_syscall_error.type = seL4_RangeError;
         current_syscall_error.rangeErrorMin = 0;
         current_syscall_error.rangeErrorMax = IOAPIC_IRQ_LINES - 1;
```

Comment:
Should this rangeErrorMax return value also be updated?

Reply:
I think you're right.  Will adjust.

Reply:
Done.